From 2554656ccc658af62db886beb21e9c7e246b97f9 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Fri, 14 Jun 2019 13:52:56 +0800
Subject: [PATCH 21/52] MLK-22112-3 media: mipi_csi: replace reset flow for phy

After dispmix reset driver has been introduced, use reset control framework
to do the PHY reset for mipi csi.

BuildInfo:
 - ATF fecb1ef

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 2f6744d6ba2045b51846fe6b8b280b854a12c475)
---
 .../media/platform/mxc/capture/mxc_mipi_csi.c | 22 +++++++------------
 1 file changed, 8 insertions(+), 14 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
index a7964b7b8345..1a0dd5e7f406 100644
--- a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
+++ b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
@@ -39,6 +39,7 @@
 #include <linux/pm_runtime.h>
 #include <linux/regmap.h>
 #include <linux/regulator/consumer.h>
+#include <linux/reset.h>
 #include <linux/slab.h>
 #include <linux/spinlock.h>
 #include <linux/videodev2.h>
@@ -190,9 +191,6 @@ MODULE_PARM_DESC(debug, "Debug level (0-2)");
 #define MIPI_CSIS_PKTDATA_EVEN		0x3000
 #define MIPI_CSIS_PKTDATA_SIZE		SZ_4K
 
-#define GPR_MIPI_RESET			0x08
-#define GPR_MIPI_S_RESETN		BIT(16)
-
 #define DEFAULT_SCLK_CSIS_FREQ	166000000UL
 
 enum {
@@ -393,23 +391,19 @@ static int mipi_csis_phy_init(struct csi_state *state)
 
 static int mipi_csis_phy_reset_mx8mm(struct csi_state *state)
 {
-	struct device_node *np = state->dev->of_node;
-	struct regmap *gpr;
+	struct reset_control *phy_reset;
 
-	gpr = syscon_regmap_lookup_by_phandle(np, "csi-gpr");
-	if (IS_ERR(gpr))
-		return PTR_ERR(gpr);
+	phy_reset = devm_reset_control_get_exclusive(state->dev, "csi,mipi_rst");
+	if (IS_ERR(phy_reset))
+		return PTR_ERR(phy_reset);
 
-	regmap_update_bits(gpr, GPR_MIPI_RESET,
-			   GPR_MIPI_S_RESETN,
-			   0x0);
+	reset_control_assert(phy_reset);
 	usleep_range(10, 20);
-	regmap_update_bits(gpr, GPR_MIPI_RESET,
-			   GPR_MIPI_S_RESETN,
-			   GPR_MIPI_S_RESETN);
+	reset_control_deassert(phy_reset);
 	usleep_range(10, 20);
 
 	return 0;
+
 }
 
 static int mipi_csis_phy_reset(struct csi_state *state)
-- 
2.17.1

